Autopilot/PathManager/Inc/pathStateClasses.hpp

```diff
@@ -0,0 +1,120 @@
+ * Code
+ **********************************************************************************************************************/
+
+class commsWithAttitude : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        commsWithAttitude() {}
+        commsWithAttitude(const commsWithAttitude& other);
+        commsWithAttitude& operator =(const commsWithAttitude& other);
+};
+
+class getFromTelemetry : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        getFromTelemetry() {}
+        getFromTelemetry(const getFromTelemetry& other);
+        getFromTelemetry& operator =(const getFromTelemetry& other);
+};
+
+class getSensorData : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static AltimeterData_t *GetAltimeterOutput(void) {return &_altimeterdata;}
+        static GpsData_t *GetGPSOutput(void) {return &_gpsdata;}
+    private:
+        getSensorData() {}
+        getSensorData(const getSensorData& other);
+        getSensorData& operator =(const getSensorData& other);
+        static AltimeterData_t _altimeterdata; 
+        static GpsData_t _gpsdata; 
+};
+
+class sensorFusion : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        sensorFusion() {}
+        sensorFusion(const sensorFusion& other);
+        sensorFusion& operator =(const sensorFusion& other);
+};
+
+class cruisingState : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static _PathData *GetWaypointData(void) {return &_waypointdata;}
+        static _WaypointManager_Data_Out *GetOutputData(void) {return &_outputdata;}
+    private:
+        cruisingState() {}
+        cruisingState(const cruisingState& other);
+        cruisingState& operator =(const cruisingState& other);
+        //WaypointManager(0,0) {} initializing instance of WaypointManager class
+        int waypointIDArray[PATH_BUFFER_SIZE]; 
+        static _PathData _waypointdata; 
```

Comment:
You don't need any instances of the _PathData struct inside of the cruisingState state. If we need to create them, they should be made in the execute() method when they are called